content/blog/dtplyr-1-3-0/index.Rmd

```diff
@@ -0,0 +1,102 @@
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+```
+We're thrilled to announce the release of [dtplyr](https://dtplyr.tidyverse.org) 1.3.0.
+dtplyr gives you the speed of [data.table](http://r-datatable.com/) with the syntax of dplyr; you write dplyr (and tidyr) code and dtplyr translates it to the data.table equivalent.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dtplyr")
+```
+
+This blog post will give you an overview of the changes in this version: dtplyr no longer adds translations directly to data.tables, some dplyr 1.1.0 updates, and some performance improvements.
+As always, you can see a full list of changes in the [release notes](https://github.com/tidyverse/dtplyr/releases/tag/v1.3.0)
+
+```{r setup}
+library(dtplyr)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## Breaking changes
+
+In previous versions, dtplyr registered translations that kicked in whenever you used a data.table.
+This [caused problems](https://github.com/tidyverse/dtplyr/issues/312) because merely loading dtplyr could cause otherwise ok code to fail because dplyr and tidyr functions would now return `lazy_dt` objects instead of `data.table` objects.
+To avoid this problem, we have removed those S3 methods and now must explicitly opt-in to dtplyr translations by using `lazy_dt()`.
+
+## dplyr 1.1.0
+
+This release brings support for dplyr 1.1.0's [per-operation grouping](https://www.tidyverse.org/blog/2023/02/dplyr-1-1-0-per-operation-grouping/) and `pick()`:
+
+```{r}
+dt <- lazy_dt(data.frame(x = 1:10, id = 1:2))
+dt |> 
+  summarise(mean = mean(x), .by = id) |> 
+  show_query()
+
+dt <- lazy_dt(data.frame(x = 1:10, y = runif(10)))
+dt |> 
+  mutate(row_sum = rowSums(pick(x))) |> 
+  show_query()
+```
+
+Per-operation grouping was one of the dplyr 1.1.0 features inspired by data.table, so it's neat to see it come full circle in this dtplyr release.
+Future releases will add support for other dplyr 1.1.0 features like the new [`join_by()`](https://www.tidyverse.org/blog/2023/01/dplyr-1-1-0-joins/#join_by) syntax and [`reframe()`](https://www.tidyverse.org/blog/2023/02/dplyr-1-1-0-pick-reframe-arrange/#reframe).
+
+## Improved translations
+
+dtplyr gains new translations for `add_count()` and `unite()`, and the ranking functions, `min_rank()`, `dense_rank()`, `percent_rank()`, & `cume_dist()` are now mapped to their `data.table` equivalents:
+
+```{r}
+dt |> add_count() |> show_query()
+
+dt |> tidyr::unite("z", c(x, y)) |> show_query()
+
+dt |> mutate(r = min_rank(x)) |> show_query()
+
+dt |> mutate(r = dense_rank(x)) |> show_query()
+```
+
+This release also includes three translation improvements that yield better performance:
+
+-   Where possible, `arrange()` will use `setorder()` instead of `order()`.
```

Comment:
Maybe it would be useful to clarify when these alternate translations will be used?

```
-   When data has previously been copied:

    * `arrange()` will use `setorder()` instead of `order()`.
    
    * `select()` will drop unwanted columns by reference (i.e. with `var := NULL`).
 ```